stdlib/universe/parallel.go

```diff
@@ -0,0 +1,168 @@
+package universe
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sync"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/plan"
+	"github.com/opentracing/opentracing-go"
+)
+
+const (
+	PartitionMergeKind = "influx2x/partitionMergeKind"
+)
+
+type PartitionMergeProcedureSpec struct {
+	plan.DefaultCost
+}
+
+func (o *PartitionMergeProcedureSpec) Kind() plan.ProcedureKind {
+	return PartitionMergeKind
+}
+
+func (o *PartitionMergeProcedureSpec) Copy() plan.ProcedureSpec {
+	return &PartitionMergeProcedureSpec{
+		DefaultCost: o.DefaultCost,
+	}
+}
+
+func init() {
+	execute.RegisterTransformation(PartitionMergeKind, createPartitionMergeTransformation)
+}
+
+func createPartitionMergeTransformation(id execute.DatasetID, mode execute.AccumulationMode, spec plan.ProcedureSpec, a execute.Administration) (execute.Transformation, execute.Dataset, error) {
+	s, ok := spec.(*PartitionMergeProcedureSpec)
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid spec type %T", spec)
+	}
+
+	alloc := a.Allocator()
+
+	d := execute.NewPassthroughDataset(id)
+
+	t, err := NewPartitionMergeTransformation(a.Context(), d, alloc, s, a.Parents())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return t, d, nil
+}
+
+type PartitionMergeTransformation struct {
+	execute.ExecutionNode
+	ctx     context.Context
+	dataset *execute.PassthroughDataset
+	span    opentracing.Span
+	alloc   *memory.Allocator
+
+	mu          sync.Mutex
+	parentState map[execute.DatasetID]*parallelParentState
+}
+
+type parallelParentState struct {
```

Comment:
Could we change `Parent` to `Predecessor` here? I find the former to be ambiguous and confusing in a DAG.

Reply:
Done.